rare/components/tabs/games/game_utils.py

```diff
@@ -15,26 +16,70 @@
 from rare.components.tabs.games import CloudSaveUtils
 from rare.shared import LegendaryCoreSingleton, GlobalSignalsSingleton, ArgumentsSingleton
 from rare.utils import legendary_utils
+from rare.utils import utils
 from rare.utils.meta import RareGameMeta
 
 logger = getLogger("GameUtils")
 
 
-class GameProcess(QProcess):
+class GameProcess(QObject):
+    @dataclass
+    class FinishedModel:
+        action: str
+        app_name: str
+        exit_code: int
+        playtime: int  # seconds
+
+        @classmethod
+        def from_json(cls, data):
+            return cls(
+                action=data["action"],
+                app_name=data["app_name"],
+                exit_code=data["exit_code"],
+                playtime=data["playtime"],
+            )
+
     game_finished = pyqtSignal(int, str)
 
-    # noinspection PyUnresolvedReferences
-    def __init__(self, app_name):
+    def __init__(self, app_name: str):
         super(GameProcess, self).__init__()
         self.app_name = app_name
-        self.finished.connect(self._game_finished)
-
-    def _game_finished(self, exit_code):
+        self.socket = QLocalSocket()
+        self.socket.connected.connect(self._socket_connected)
+        self.socket.errorOccurred.connect(self._error_occured)
+        self.socket.readyRead.connect(self._message_available)
+        self.socket.disconnected.connect(lambda: self.socket.close())
+
+        # wait a short time for process started
+        QTimer.singleShot(1000, self.connect_to_server)
```

Comment:
How about a recurring timer with a smaller wait-time? It can be disabled in `connect_to_server`

Reply:
It would feel a bit like spam, but the idea is good. I thought, if the process does not start after a second, it failed to start

Reply:
With a good value, like 200ms it shouldn't be too spammy, and if the connection was successful it get disabled afterwards